qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -205,6 +205,86 @@ def _mcsu2_real_diagonal(
     return circuit
 
 
+def _mcsu2_with_params(
+    unitary_type: str,  #'rz', 'ry' or 'rz'
+    theta: ParameterValueType,
+    num_controls: int,
+    ctrl_state: Optional[str] = None,
+) -> QuantumCircuit:
+    """
+    Return a parametrized multi-controlled RX or RY or RZ gate [1]_.
+
+    Args:
+        unitary_type: SU(2) unitary matrix type: 'rx' or 'ry' or 'rz'.
+        num_controls: The number of control qubits.
+        ctrl_state: The state on which the SU(2) operation is controlled. Defaults to all
+            control qubits being in state 1.
+
+    Returns:
+        A :class:`.QuantumCircuit` implementing the multi-controlled RX / RY / RZ gate.
+
+    References:
+
+        .. [1]: R. Vale et al. Decomposition of Multi-controlled Special Unitary Single-Qubit Gates
+            `arXiv:2302.06377 (2023) <https://arxiv.org/abs/2302.06377>`__
+
+    """
+    from .x import MCXVChain
+    from qiskit.circuit.library.standard_gates.ry import RYGate
+    from qiskit.circuit.library.standard_gates.rz import RZGate
+
+    if unitary_type == "ry":
+        s_gate = RYGate(-theta / 4)
+    elif unitary_type in ("rz", "rx"):
+        s_gate = RZGate(-theta / 4)
+    else:
+        raise QiskitError("The unitary given is not of type rx, ry or rz.")
+
+    k_1 = math.ceil(num_controls / 2.0)
+    k_2 = math.floor(num_controls / 2.0)
+
+    ctrl_state_k_1 = None
+    ctrl_state_k_2 = None
+
+    if ctrl_state is not None:
+        str_ctrl_state = f"{ctrl_state:0{num_controls}b}"
+        ctrl_state_k_1 = str_ctrl_state[::-1][:k_1][::-1]
+        ctrl_state_k_2 = str_ctrl_state[::-1][k_1:][::-1]
+
+    circuit = QuantumCircuit(num_controls + 1, name="MCSU2")
+    controls = list(range(num_controls))  # control indices, defined for code legibility
+    target = num_controls  # target index, defined for code legibility
+
+    if unitary_type == "rx":
+        circuit.h(target)
+
+    mcx_1 = MCXVChain(num_ctrl_qubits=k_1, dirty_ancillas=True, ctrl_state=ctrl_state_k_1)
```

Comment:
Just a reminder from our offline discussions that these should use the non-deprecated synthesis methods 🙂

Reply:
Yes. I also plan to move this code to the synthesis library.